GooglePay/src/main/java/com/braintreepayments/api/googlepay/GooglePayRequest.kt

```diff
@@ -0,0 +1,437 @@
+package com.braintreepayments.api.googlepay
+
+import android.os.Parcel
+import android.os.Parcelable
+import android.text.TextUtils
+import com.google.android.gms.wallet.ShippingAddressRequirements
+import com.google.android.gms.wallet.TransactionInfo
+import com.google.android.gms.wallet.WalletConstants
+import com.google.android.gms.wallet.WalletConstants.BillingAddressFormat
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.Locale
+
+/**
+ * Represents the parameters that are needed to use the Google Pay API.
+ */
+@SuppressWarnings("TooManyFunctions")
+class GooglePayRequest : Parcelable {
```

Comment:
Can we use the parcelize annotation instead of Parcelable here? then we shouldn't need the implementation code

Reply:
I did not use Parcelize annotation was because the internal HashMap properties have values with type of JSONObject and JSONArray, they do not implement Parcelable. However, I just found they are not write to parcel either, are they intended to be omitted?

Reply:
The `:GooglePay` module definitely has some tech debt associated with it. We found a bug a while back where a critical property [wasn't being parceled properly](https://github.com/braintree/braintree_android/pull/705/files), so it wouldn't surprise me if the parcelable implementation is lacking.

The error-prone nature of maintaining our own `Parcelable` implementation highlights the huge benefit we gain by moving to `@Parcelize`. I don't know if it comes in a separate PR, but if we need to refactor `GooglePayRequest` I think we should.

Reply:
Without making a major change (keep the public functions signatures unchanged), I am thinking to change the HashMap value type from JSONObject/JSONArray to String. Then there will be a data conversion in the get and set functions. For example, 

   private val allowedPaymentMethods = HashMap<String, String>()

    fun getAllowedPaymentMethod(type: String): JSONObject? {
        return jsonObjectParser.parse(allowedPaymentMethods[type])
    }

    fun setAllowedPaymentMethod(paymentMethodType: String, parameters: JSONObject) {
        allowedPaymentMethods[paymentMethodType] = parameters.toString()
    }

Let me know what you think.

Reply:
I'm not sure immediately. Personally I'd prefer to get the Kotlin changes in and maybe we tackle in a separate PR? We're on a tight timeline to move out of beta and into GA. There may be an opportunity to for us to refactor in a minor version bump.

Reply:
I made a change in the last commit already, https://github.com/braintree/braintree_android/pull/1114/commits/4b22de09555202a480a92ad211254fd7c0642b15

Reply:
Oh that's fine. We may want to revisit after still. This is the only merchant facing API that I'm aware of that takes `JSONObject` parameters. It would be nice to fully encapsulate our JSON serialization preference in case we decide to change to Gson one day.